lib/api/v3/work_packages/schema/work_package_schema.rb

```diff
@@ -54,14 +54,21 @@ def assignable_statuses_for(user)
 
             status_origin = @work_package
 
-            # do not allow to skip statuses without intermediate saving
+            # do not allow to skip statuses without intermediately saving the work package
+            # we therefore take the original status of the work_package, while preserving all
+            # other changes to it (e.g. type, assignee, etc.)
             if @work_package.persisted? && @work_package.status_id_changed?
-              status_origin = @work_package.class.find(@work_package.id)
+              status_origin = @work_package.clone
+              status_origin.status = WorkPackage.find(@work_package.id).status
```

Comment:
@NobodysNightmare if you only need the status you could fetch it directly:
`status_origin.status = Status.find_by_id(@work_package.status_id)`
That would reduce the amount of objects to be loaded. On the other hand it will include another dependency within the code (the constant called `Status`) but I think that is acceptable.